server/vulnerabilities/macoffice/parser.go

```diff
@@ -33,6 +34,52 @@ var nameToType = map[string]ProductType{
 	"microsoft autoupdate": WholeSuite,
 }
 
+type releaseNotesURL string // type added to calm gosec variable URL warnings down
+const (
+	primaryURL = releaseNotesURL("https://learn.microsoft.com/en-us/officeupdates/release-notes-office-for-mac")
+	archiveURL = releaseNotesURL("https://learn.microsoft.com/en-us/officeupdates/release-notes-office-for-mac-archived")
+)
+
+func GetReleaseNotes(includeInvalid bool) (ReleaseNotes, error) {
+	var relNotes ReleaseNotes
+
+	relNotes, err := addReleaseNotes(relNotes, primaryURL, includeInvalid)
+	if err != nil {
+		return nil, err
+	}
+
+	relNotes, err = addReleaseNotes(relNotes, archiveURL, includeInvalid)
+	if err != nil {
+		return nil, err
+	}
+
+	return relNotes, nil
+}
+
+func addReleaseNotes(relNotes ReleaseNotes, url releaseNotesURL, includeInvalid bool) (ReleaseNotes, error) {
+	res, err := http.Get(string(url))
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	parsed, err := parseReleaseHTML(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rn := range parsed {
+		// Under normal operation (outside parser tests), we only care about release notes that have a version set
+		// (because we need that for matching software entries) and also that contain some
+		// security updates (because we only intend to use the release notes for vulnerability processing).
+		if includeInvalid || rn.Valid() {
+			relNotes = append(relNotes, rn)
+		}
```

Comment:
Just so I understand correctly, we need the `includeInvalid` for the parser tests exclusively? I assume we are doing this because it is difficult / not possible to mock the `Valid()` method in the parser tests?
[Not a blocker]

Reply:
That's correct. Previously the tests didn't test that code path at all...they pulled stuff over HTTP independently because the HTTP-pulling machinery was in main() on the command. That meant that the validation bit got skipped, but it also meant that test coverage was lower of the code path the real command was using. The refactor alongside the bugfix increases code coverage, but without the ability to skip validation the expected test values no longer match what we hand back, and I didn't want to make the diff bigger than it already was.